### PR TITLE
Unity Android Internet Identity integration

### PR DESCRIPTION
**Pull request by vincent-dfinity**
This sample shows one way to integrate Internet Identity with Unity Android apps. It contains two examples: one for deeplink, the other one for app links. Each of the example contains: a dapp with II integrated, and an Unity Project which interacts with this dapp.

The basic workflow:
1. Set up an Internet Identity integration dapp which supports logging in with II.
2. Run a Unity game on Android.
3. Launch the Web Browser from the game to open the dapp frontend deployed in step 1, with the public key of `Ed25519Identity` as a parameter.
4. Login with your Internet Identity in the Web Browser.
5. Launch the application via DeepLink or App Links, and pass the `DelegationIdentity` back to the game as the URL parameter.
6. Call the backend canister with the `DelegationIdentity` to greet through [ICP.NET](https://github.com/BoomDAO/ICP.NET).

Here is a demo link for [II + Unity Android App Links](https://drive.google.com/file/d/1iw4AUPTvCmxO7mMPK9miPAGGvd9viIK7/view?usp=sharing ).

### CODE REVIEW

File: native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const openButton = document.getElementById("open");
+openButton.onclick = async (e) => {
+    e.preventDefault();
+
+    if (sessionKeyIndex === -1) {
+        // TODO: warning for not login from a game.
+        return false;
+    }
+    
+    var url = "internetidentity://authorize?";
```
Comment (frederikrothenberger) [inline native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js:73]: This is an app-link, correct?

File: native-apps/unity_android_applink/ii_integration_page/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+With this, users don't need to pass the private key around, also they don't need to store the private key outside of the game as they can regenerate the key pairs for every session.
+
+## Deployment
+
+If you never work on the IC and don't know how to deploy a dapp, please refer to the [Hello World sample](https://internetcomputer.org/docs/current/tutorials/deploy_sample_app) to learn the basic knowledge about the IC. 
+
+### Set up the `assetlinks.json` file
+
+In order to support Android App Links, the website needs to serve an `assetlinks.json` file under the `.well-known` directory. The `assetlinks.json` file contains the package name, sha256 certificate fingerprints etc., which will be used by the Android installer to verify if the apps should be launched to handle the URLs defined in the Android manifest.xml. This is an example of [assetlinks.json](./src/greet_frontend/assets/.well-known/assetlinks.json) file in this sample.
```
Comment (frederikrothenberger) [inline native-apps/unity_android_applink/ii_integration_page/README.md:23]: The file `./src/greet_frontend/assets/.well-known/assetlinks.json` seems to be missing in this example?

**Review by frederikrothenberger — COMMENTED** [review]
Hi @vincent-dfinity 

Thanks for creating this example! Unfortunately, I don't have the time for a thorough review right now, but from a first glance it looks really good, especially that you looked into the app-link integration. 👍 

I would suggest pinging prod-sec to do the review of the II integration.

File: native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js (hunk 1 above)
Comment (vincent-dfinity) [reply native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js:73]: This is a deep link, you can customize the scheme as you wish.

And for app links, per https://developer.android.com/training/app-links/, you can only use HTTP or HTTPS schemes.

File: native-apps/unity_android_applink/ii_integration_page/README.md (hunk 2 above)
Comment (vincent-dfinity) [reply native-apps/unity_android_applink/ii_integration_page/README.md:23]: It's under the `./src/greet_frontend/assets/.well-known/assetlinks.json` in the same directory of this README.md.

Here it is https://github.com/dfinity/examples/blob/unity-ii-integration/native-apps/unity_android_applink/ii_integration_page/src/greet_frontend/assets/.well-known/assetlinks.json

Comment (frederikrothenberger) [reply native-apps/unity_android_applink/ii_integration_page/README.md:23]: Ah, sorry. I did not realize that you did 2 examples for both ways of integrating. 👍 
Thanks!

File: native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js (hunk 1 above)
Comment (frederikrothenberger) [reply native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js:73]: See other comment, I did not realize there are two types of integration. Perfect 🙂

**Pull request merged**